tinyiiod.c

```diff
@@ -161,7 +161,8 @@ void tinyiiod_do_write_attr(struct tinyiiod *iiod, const char *device,
 		bytes = sizeof(buf);
 
 	tinyiiod_read(iiod, buf, bytes);
-	buf[bytes] = '\0';
+	if(bytes < sizeof(buf))
```

Comment:
this doesn't look correct; from what i can deduce, the null-char is important ;
so, if you don't set it, then a string won't be null-terminated;

even though, it isn't great, i suggest cutting `bytes` by 1 with
```
	if (bytes > sizeof(buf) - 1)
		bytes = sizeof(buf) - 1;
```

this would at least give you a null-terminated string

Reply:
The upper implementation is used now